### PR TITLE
ENH: Dtype._is_immutable

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @phofl

### CODE REVIEW

**Review by phofl — APPROVED** [review]
lgtm

**Comment by phofl** [comment]
thx @jbrockmendel

File: pandas/core/dtypes/base.py [hunk 1]
```diff
@@ -396,6 +396,16 @@ def _can_hold_na(self) -> bool:
         """
         return True
 
+    @property
+    def _is_immutable(self) -> bool:
+        """
+        Can arrays with this dtype be modified with __setitem__? If not, return
```
Comment (mroeschke) [inline pandas/core/dtypes/base.py:402]: Should `ExtensionArray.__setitem__` have a `if dtype._is_immutable: raise TypeError` check to solidify this?

Comment (jbrockmendel) [reply pandas/core/dtypes/base.py:402]: I guess that would be harmless since any working `EA.__setitem__` would need to override

**Pull request merged**